### ISSUE TITLE
Corrupted JPEG,I can't solved.

### ISSUE THREAD

**Issue by 992060356** [ultralytics/yolov5#3731]
## ❔Question
Why is my image or label corrupted? A lot of pictures are corrupted, so I checked some of them and opened the pictures and the corresponding label. There is no problem.
Can you give me some help?
I tried to delete cacha and retrain
I looked at utiles / datasets.py, but I didn't know how to debug it.

## Additional context
`train: WARNING: Ignoring corrupted image and/or label voc/images/9556041603720_camera3-8.jpg: corrupted JPEG
train: WARNING: Ignoring corrupted image and/or label voc/images/9556041603720_camera3-9.jpg: corrupted JPEG
train: Scanning 'voc/train' images and labels...29281 found, 0 missing, 0 empty, 46085 corrupted: 100%|█| 75366/75366 [00:08<00
train: New cache created: voc/train.cache

val: WARNING: Ignoring corrupted image and/or label voc/images/9556041603720_camera2-25.jpg: corrupted JPEG
val: WARNING: Ignoring corrupted image and/or label voc/images/6907777808584-back_camera2-20.jpg: corrupted JPEG
val: Scanning 'voc/val' images and labels...333 found, 0 missing, 0 empty, 505 corrupted: 100%|███████| 838/838 [00:00<00:00, 1
val: New cache created: voc/val.cache
Plotting labels...

autoanchor: Analyzing anchors... anchors/target = 3.51, Best Possible Recall (BPR) = 1.0000
Image sizes 1280 train, 1280 test
Using 4 dataloader workers
Logging results to runs/train/RPC
Starting training for 600 epochs...

     Epoch   gpu_mem       box       obj       cls     total    labels  img_size
     0/599     20.3G   0.06935    0.1463    0.1145    0.3301        72      1280:  37%|█████         | 2673/7321 [37:15<1:03:24     0/599     20.3G   0.06935    0.1463    0.1145    0.3301        72      1280:  37%|█████         | 2673/7321 [37:15<1:04:47
`

**Comment by tetelevm** [comment]
What are your image and label paths? Also, what does a typical labels file look like?

**Comment by glenn-jocher** [comment]
@992060356 if you believe your JPEGs are ok then you may want to load and save them with cv2 so that they are properly saved, avoiding the corrupt JPEG message.

**Comment by SpongeBab** [comment]
Are you sure?Please check them carefully again.May be this will help you.#3656

**Comment by 992060356** [comment]
> What are your image and label paths? Also, what does a typical labels file look like?
this is my yaml config file.
`# train and val data as 1) directory: path/images/, 2) file: path/images.txt, or 3) list: [path1/images/, path2/images/]
train: voc/train.txt  # 75366 images
val: voc/val.txt  # 838 images
test: voc/test.txt  # 7535 images`
The image path is in the txt file.
I found a corrupted image's label, which looks like this.
`111 0.5335648148148148 0.6144547325102881 0.15046296296296297 0.14660493827160495`
this is image
![038900004095_camera0-13](https://user-images.githubusercontent.com/75258166/123021233-4b1cc100-d406-11eb-9131-e54716a14c8b.jpg)

**Comment by 992060356** [comment]
> @992060356如果您认为 JPEG 没问题，那么您可能希望使用 cv2 加载和保存它们，以便正确保存它们，避免损坏的 JPEG 消息。



> @992060356如果您认为 JPEG 没问题，那么您可能希望使用 cv2 加载和保存它们，以便正确保存它们，避免损坏的 JPEG 消息。

thanks  for your reply，I will try this way.

**Comment by 992060356** [comment]
> Are you sure?Please check them carefully again.May be this will help you.#3656

**Comment by 992060356** [comment]
> 你确定吗？请再仔细检查一下。也许这对你有帮助。#3656

看了你的#3656问题，我跟你的问题不太一样，标签中的坐标也没有越界。

**Comment by 992060356** [comment]
> @992060356如果您认为 JPEG 没问题，那么您可能希望使用 cv2 加载和保存它们，以便正确保存它们，避免损坏的 JPEG 消息。

I'm realy happy，I take cv2 load and save them ,then try again ,it's successfull, why? I looked at their properties，It is found that the image file is more than 50 K larger than the original image file, and has more horizontal and vertical resolution attributes

**Comment by glenn-jocher** [comment]
@992060356 @SpongeBab one option here might be to simply re-save the corrupted jpegs with cv2 automatically, or provide an option to do this. 

But in the case of seriously problematic JPEGs this would allow them to be included in training.

**Comment by 992060356** [comment]
> @992060356 @SpongeBab one option here might be to simply re-save the corrupted jpegs with cv2 automatically, or provide an option to do this.
> 
> But in the case of seriously problematic JPEGs this would allow them to be included in training.

Thank you very much for your suggestions, have not found the specific reasons, I do not know if there is a relationship with the cv2 version

**Comment by ccl-private** [comment]
I met this problem too. why this is closed?
I write images using cv2, and my cv2 version is 4.5.2.54.

**Comment by ccl-private** [comment]
OH! duplicate labels.
my fault.

**Comment by glenn-jocher** [comment]
@992060356 @ccl-private good news 😃! Your original issue may now be fixed ✅ in PR #4548. This PR automatically restores and saves corrupted JPEGs, and all images are now used for training, including the restored JPEGs.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by apoorv4wadhwani** [comment]
> @992060356 @ccl-private good news 😃! Your original issue may now be fixed ✅ in PR #4548. This PR automatically restores and saves corrupted JPEGs, and all images are now used for training, including the restored JPEGs.
> 
> To receive this update:
> 
> * **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
> * **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
> * **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolov5)
> * **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image [![Docker Pulls](https://camo.githubusercontent.com/280faedaf431e4c0c24fdb30ec00a66d627404e5c4c498210d3f014dd58c2c7e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)
> 
> Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

I think this should not have happened, I had no idea that running the training routine will change the files that are stored on the computer. I think this is a weird side effect that should be optional and by default not allowed.

**Comment by 992060356** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
